server/routers/files/files_router.py

```
@@ -214,46 +216,103 @@ async def create_upload_files(
+            await asyncio.to_thread(invalidate_cdn)
+
+        upload_jobs.finish_job(album_slug)
+        await _notify("done", image_count, image_count)
+    except Exception as e:
+        print(f"album processing failed for {album_slug}: {e}")
+        upload_jobs.fail_job(album_slug, str(e))
+        await _notify("error", 0, 0)
+
+
+@router.get("/api/upload-status/{album_slug}")
+async def upload_status(album_slug: str, current_user=Depends(require_admin)):
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)